certora/specs/GuardianStorage.spec

```diff
@@ -0,0 +1,378 @@
+/*
+ * Spec for linked list reachability for Guardians.
+ * This file is derived from OwnerReach.spec for a Safe account.
+ * https://github.com/safe-global/safe-smart-account/blob/main/certora/specs/OwnerReach.spec
+ *
+ * This file uses a reach predicate:
+ *    ghost reach(address, address, address) returns bool
+ * to represent the transitive of the next
+ * relation given by the "owners" field relation for a specific wallet.
+ *
+ * The idea comes from the paper
+ *
+ * [1] Itzhaky, S., Banerjee, A., Immerman, N., Nanevski, A., Sagiv, M. (2013).
+ *     Effectively-Propositional Reasoning about Reachability in Linked Data Structures.
+ *     In: CAV 2013. Springer, https://doi.org/10.1007/978-3-642-39799-8_53
+ */
+
+using SafeHarness as safeContract;
+using SocialRecoveryModule as socialRecoveryModule;
+
+methods {
+    // Safe Functions
+    function safeContract.isModuleEnabled(address) external returns (bool) envfree;
+
+    // GuardianManager Functions
+    function threshold(address) external returns (uint256) envfree;
+    function isGuardian(address, address) external returns (bool) envfree;
+
+}
+
+definition MAX_UINT256() returns uint256 = 0xffffffffffffffffffffffffffffffff;
```

Comment:
I learned recently that certora has built-in constants for max values: 
https://forum.certora.com/t/is-there-a-pre-defined-max-value-constant-in-cvl/39/2

So this could be replaced with `max_uint256` built-in constant

Reply:
https://github.com/5afe/CandideWalletContracts/pull/20/commits/6447b22c0e01d8ef82cfa79617354991df479e63